### PR TITLE
Jsonata sources

### PR DESCRIPTION
**Pull request by qbalukom**
Introduce `source` instead of `name` and `formula` for display widgets.
Unify data sources for all widgets (e.g. `ResourceList`).
Validate details using schema instead of hard-coded rules.

### CODE REVIEW

File: core/src/luigi-config/customResources.js [hunk 1]
```diff
@@ -123,3 +123,15 @@ export async function getCustomResources(authData) {
   }
   return [];
 }
+
+export async function getSchemas() {
```
Comment (dariadomagala) [inline core/src/luigi-config/customResources.js:127]: Can we rename it? For me, the name suggests that we're taking all the CRDs schemas, while we're taking our own extensibility schemas

File: docs/extensibility/display-widgets.md [hunk 2]
```diff
@@ -285,7 +285,7 @@ If such resource list was already defined in Busola, the configuration will be r
   "widget": "ResourceList",
   "path": "$myRelatedResource",
   "name": "Example ResourceList Secret",
-  "columns": [
+  "children": [
```
Comment (dariadomagala) [inline docs/extensibility/display-widgets.md:288]: what about adjusting the examples?

File: core-ui/src/components/Extensibility/ExtensibilityDetails.js [hunk 3]
```diff
@@ -23,6 +30,11 @@ export const ExtensibilityDetailsCore = ({ resMetaData }) => {
 
   const detailsProps = usePrepareDetailsProps(urlPath, 'name');
 
+  const ajv = new Ajv();
+  if (!ajv.validate(extSchemas.details, resMetaData?.details)) {
+    throwConfigError(t('extensibility.errors'), { error: ajv.errors });
```
Comment (dariadomagala) [inline core-ui/src/components/Extensibility/ExtensibilityDetails.js:35]: 1. `t` here comes from extensibility translations, not our busola ones so it won't work
2. there's no `extensibility.errors` in our translations, you can use `extensibility.error` (without s) instead, but that will only double the same error. Please use something more descriptive.

**Review by dariadomagala — CHANGES_REQUESTED** [review]

**Review by dariadomagala — CHANGES_REQUESTED** [review]
1. Translations are missing now:
Screen Shot 2022-08-17 at 11 15 59
or 
Screen Shot 2022-08-17 at 11 25 27

2. I've chosen a pizza's secret in the form, but it's not visible in the details
Screen Shot 2022-08-17 at 11 25 53

File: core-ui/src/components/Extensibility/helpers.js [hunk 4]
```diff
@@ -36,25 +38,44 @@ export const useGetTranslation = path => {
...
+      defaultValue: value,
+    });
+  };
+
+  const tFromStoreKeys = (storeKeys, schema, options) => {
+    return widgetT({
+      ...schema.toJS(),
+      path: storeKeys.toArray().filter(el => typeof el === 'string'),
+    });
+    // const path = storeKeys
```
Comment (dariadomagala) [inline core-ui/src/components/Extensibility/helpers.js:66]: please remove unused code

**Review by dariadomagala — APPROVED** [review]

File: docs/extensibility/resources.md [hunk 5]
```diff
@@ -176,68 +171,77 @@ Extra parameters might be available for specific widgets.
...
 
+will render `spec.data[].name` and `spec.data[].description`.
+
 ## _dataSources_ section
 
-The **dataSources** section contains an object that maps a data source name to a data source configuration object. The data source name preceded by a dollar sign '\$' is used in the **path** expression.
+The **dataSources** section contains an object that maps a data source name to a data source configuration object. The data source name preceded by a dollar sign '\$' is used in the **source** expression.
 
-It's possible to use both data source name and a path; for example, `{"path": $myRelatedResource.metadata.labels}` returns the `metadata.labels` of the related resource.
+Data sources are provided in all [JSONata](https://docs.jsonata.org/overview.html) formulas as functions to call.For example, `{ "source": $myRelatedResource().metadata.labels }` returns the `metadata.labels` of the related resource.
```
Comment (grego952) [inline docs/extensibility/resources.md:242]:
```suggestion
Data sources are provided in all [JSONata](https://docs.jsonata.org/overview.html) formulas as functions to call. For example, `{ "source": $myRelatedResource().metadata.labels }` returns the `metadata.labels` of the related resource.
```

File: docs/extensibility/resources.md [hunk 6]
```diff
@@ -291,7 +295,9 @@ Those fields are used to build the related resource URL and filter the received
 
 ## _translations_ section
 
-This section contains all available languages formatted for i18next either as YAML or JSON, based on their paths.
+This optional section contains all available languages formatted for [i18next](https://www.i18next.com/) either as YAML or JSON, based on their paths. When a name is provided for a widget that value can be used as the key, and the value is the translation for a specific language.
```
Comment (grego952) [inline docs/extensibility/resources.md:298]:
```suggestion
This optional section contains all available languages formatted for [i18next](https://www.i18next.com/) either as YAML or JSON, based on their paths. When a name is provided for a widget, that value can be used as the key, and the value is the translation for a specific language.
```

File: docs/extensibility/resources.md [hunk 7]
```diff
@@ -291,7 +295,9 @@ Those fields are used to build the related resource URL and filter the received
...
+
+In addition, if no name is provided, form widgets will automatically try to fetch a translation based on it's **path** attribute, and if that fails, they will use a prettified version of it's last path item as it's name (for example `spec.itemDescription` will be prettified to "Item Description"), and by extension as a potential translation key.
```
Comment (grego952) [inline docs/extensibility/resources.md:300]:
```suggestion
In addition, if no name is provided, form widgets automatically try to fetch a translation based on their **path** attribute, and if that fails, they use a prettified version of the last path item as their name (for example `spec.itemDescription` is prettified to "Item Description"), and by extension as a potential translation key.
```

**Review by grego952 — APPROVED** [review]

**Review by dariadomagala — APPROVED** [review]

**Review by grego952 — APPROVED** [review]

**Pull request merged**